### ISSUE TITLE
I hope the system will prompt Database password:

### ISSUE THREAD

**Issue by happyzhx** [flyway/flyway#3813]
###### Which version and edition of Flyway are you using?
9.19.4

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line

###### Which database are you using? (Type & version)
Mysql & Oracle

###### Which operating system are you using?
Linux

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)
When I run the migrations (./flyway migrate -user=lodas_dba -baselineOnMigrate=true -validateOnMigrate=false -table=T_FLYWAY_SCHEMA_HISTORY -url=jdbc:oracle:thin:@10.137.34.187:1536:ora12c), I hope the system will prompt
Database password:

![image](https://github.com/flyway/flyway/assets/32672017/24de593a-611c-43c4-bce0-1c6870f6a318)

###### What did you expect to see?
When my command line does not enter -password=xxx, I want it to prompt me for the password below, like this:
![image](https://github.com/flyway/flyway/assets/32672017/56f5bab2-7da0-456e-8ff5-9e239cf7f9f1)

###### What did you see instead?
I get this error: 

ERROR: Unable to obtain connection from database (jdbc:oracle:thin:@10.137.34.187:1536:ora12c) for user 'lodas_dba': ORA-01005: null password given; logon denied

-----------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 72000
Error Code : 1005
Message    : ORA-01005: null password given; logon denied

Caused by: java.sql.SQLException: ORA-01005: null password given; logon denied

**Comment by Barry-RG** [comment]
Thank you for your suggestion. Since flyway is more commonly used in pipelines in non-interactive environments and that no password can be a valid option, this is not something Flyway currently does not do. 

If attention increases we may revisit this.

**Comment by Barry-RG** [comment]
Upon revisiting this issue, Flyway does have a prompt mechanic. However, your URL configuration is supplying Flyway with a blank password. The `@` symbol in your JDBC url is supplying an empty username and password (of which the username is being overridden by the configuration). Please refactor your command to use `-url=jdbc:oracle:thin:10.137.34.187:1536:ora12c`

**Comment by happyzhx** [comment]
> Upon revisiting this issue, Flyway does have a prompt mechanic. However, your URL configuration is supplying Flyway with a blank password. The `@` symbol in your JDBC url is supplying an empty username and password (of which the username is being overridden by the configuration). Please refactor your command to use `-url=jdbc:oracle:thin:10.137.34.187:1536:ora12c`

Sorry, I posted a problematic picture, because I can enter the password interactively using Flyway 8.2.2, but Flyway 9.19.4 does not support this method, as shown below
![image](https://github.com/flyway/flyway/assets/32672017/40b70254-c184-4981-b3b9-32c3d9915929)

**Comment by Barry-RG** [comment]
As mentioned in my previous response, Flyway has become more controlled over passwords in URL and the `@` is stating that you are supplying an intentional empty username and password of which you are overriding the username in the command line but not the password. Please remove the `@` symbol from your url to check if that resolved your problem.

**Comment by ElementUser** [comment]
> As mentioned in my previous response, Flyway has become more controlled over passwords in URL and the `@` is stating that you are supplying an intentional empty username and password of which you are overriding the username in the command line but not the password. Please remove the `@` symbol from your url to check if that resolved your problem.

Hi @Barry-RG ,

I'm encountering the exact same issue as the original poster (@happyzhx).

I'm using Flyway 10.4.1 and I've removed the `@` from the `url` field of the Flyway configuration file as you have suggested. However, I still get the same issue when using a `.conf` file: it doesn't actually read whatever I write into the input for the `user` or `password` fields when prompted (the error keeps on mentioning that the user/password fields are `null`).

I tried migrating my settings to the modern `.toml` file format, but that just outright gives me an error & mentioned that I have to supply the user/password fields explicitly (rather than being prompted to do so)
